tests/unit/mixins/route-test.js

```diff
@@ -455,6 +455,41 @@ test('it allows overrides/manual invocations of updateInfinityModel', assert =>
   assert.equal(model.get('content.lastObject.title'), 'Tender Is the Night', 'updateInfinityModel can be invoked manually');
 });
 
+test('it returns objects fetched from the store to the afterInfinityModel method', assert => {
+  var RouteObject = Ember.Route.extend(RouteMixin, {
+    model() {
+      return this.infinityModel('item');
+    },
+    afterInfinityModel(items) {
+      return items.setEach('author', 'F. Scott Fitzgerald');
```

Comment:
could we also do a test that uses a second promise as the return?

something like:
```
afterInfinityModel(items) {
  return new Promise((resolve) => {
    resolve(items.setEach('author', 'F. Scott Fitzgerald'));
  });
}
```